zer0-os/zOS
# feat: implement unread message count functionality with matrix client

### What does this do?
- This PR implements functionality to track and update unread message counts in the application using the Matrix client. It refactors the existing event handling logic to streamline the process of receiving new messages and updates the unread count accordingly. 

### Why are we making this change?
- To ensure conversations are highlighted when there are unread messages, and the unread count is updated and reflected in the UI. 

### How do I test this?
- Send messages to a zOS user you are logged in with and check the conversation list item is highlighted, and, the unread counter in the conversation list item increments for every unread message. Once the conversation has been `read` the conversation list item should no longer be highlighted and a send receipt will be sent to the server which should set the unread count back to 0, in turn the UI should reflect this by no longer highlighting the conversation list item as `unread`.

### Key decisions and Risk Assessment:
  #### Things to consider:
  1. How will this affect security?
  1. How will this affect performance?
  1. Does this change any APIs?

DEMO

https://github.com/zer0-os/zOS/assets/39112648/1402902a-4ae5-40c7-a89e-a2400cb8e37f
